core/test/gather/session-test.js

```diff
@@ -168,23 +171,16 @@ describe('ProtocolSession', () => {
       });
     });
 
-    it('respects a timeout of infinity', async () => {
-      const sendPromise = createDecomposedPromise();
-      puppeteerSession.send = fnAny().mockReturnValue(sendPromise.promise);
-
-      session.setNextProtocolTimeout(Infinity);
-      const resultPromise = makePromiseInspectable(session.sendCommand('Page.navigate', {url: ''}));
-
-      await timers.advanceTimersByTime(100_000);
-      await flushAllTimersAndMicrotasks();
-
-      expect(resultPromise).not.toBeDone();
-
-      sendPromise.resolve('result');
-      await flushAllTimersAndMicrotasks();
+    it('rejects on error from protocol', async () => {
+      rawSend.mockRejectedValue(new Error('Url is not valid'));
+      const resultPromise = session.sendCommand('Page.navigate', {url: ''});
+      await expect(resultPromise).rejects.toThrow('Url is not valid');
+    });
 
-      expect(resultPromise).toBeDone();
-      expect(await resultPromise).toBe('result');
+    it('ignores protocol timeouts from puppeteer', async () => {
+      rawSend.mockRejectedValue(new Error(`Increase 'protocolTimeout'`));
+      const resultPromise = session.sendCommand('Page.navigate', {url: ''});
+      await expect(resultPromise).resolves.toBeUndefined();
```

Comment:
In theory this result would never happen because we would throw a `PROTOCOL_TIMEOUT` of our own in this situation. So maybe this test isn't good?